### PR TITLE
protocol_splitter containerization scripts

### PR DESCRIPTION
**Pull request by jnippula**
Add scripts to create docker image for running protocol-splitter inside container

### CODE REVIEW

File: docker_image_build.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+docker build -t protocol-splitter-cont --no-cache .
```
Comment (joonas-fi) [inline docker_image_build.sh:1]: There is nothing wrong in this file (or the other one), but philosophically both of these are instructions that IMO could be in a README like this:

```
How to run in Docker
----------------------------

'''
$ docker run --name prot_split --rm -dt --device /dev/ttyS7 protocol-splitter-cont 127.0.0.1 /dev/ttyS7
'''

How to build Docker image
------------------------------------

'''
$ docker build -t protocol-splitter-cont --no-cache .
'''
```

Rationale: usually these scripts are not used/required in production or the CI (the `.github/workflows/..` file already has its own explicit way of doing this). In production there's also an explicit spell for running it: https://github.com/tiiuae/fog_hyper/blob/37b43c44ee110491bc4571e3ce68dd944d1964f5/pkg/manifest/manifest.go#L73
=> these scripts serve the occasional developer/user who should be fine copy-pasting the advice from readme and making their own scripts if they have repetitive / exotic needs

But this is just an opinion, there is nothing wrong in these being as scripts, no need to spend time "fixing" even if you agree. Just a thought to think about in the future :)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+COPY --from=builder /lib/x86_64-linux-gnu/libpthread.so.0 /lib/x86_64-linux-gnu/
+COPY --from=builder /lib/x86_64-linux-gnu/libstdc++.so.6 /lib/x86_64-linux-gnu/
+COPY --from=builder /lib64/ld-linux-x86-64.so.2 /lib64/
+
+COPY --from=builder /agent_protocol_splitter/build/protocol_splitter /
+COPY /entrypoint.sh /
+
+ENTRYPOINT ["/entrypoint.sh"]
+
+EXPOSE 14540/udp
```
Comment (joonas-fi) [inline Dockerfile:39]: A minor nitpick (no need to fix) to keep in mind in the future: rarely-changing Docker instructions should be closer to top for improved caching and layer reuse.

`EXPOSE ...` here means that every time the code changes (`COPY --from=builder ...`), new image layers with expose instructions have to be built.

Example: expose last
----------------------------

<details>
<summary>Show</summary>

```
/tmp/dockertest
λ cat Dockerfile
FROM scratch

ADD date /date

EXPOSE 8080/tcp


/tmp/dockertest
λ echo $(date) > date && docker build -t test .
Sending build context to Docker daemon  3.072kB
Step 1/3 : FROM scratch
 --->
Step 2/3 : ADD date /date
 ---> 7cdf307f0f6d
Step 3/3 : EXPOSE 8080/tcp
 ---> Running in c48b45df3b9d
Removing intermediate container c48b45df3b9d
 ---> fbe1307569c5
Successfully built fbe1307569c5
Successfully tagged test:latest

/tmp/dockertest
λ echo $(date) > date && docker build -t test .
Sending build context to Docker daemon  3.072kB
Step 1/3 : FROM scratch
 --->
Step 2/3 : ADD date /date
 ---> 117bb067fae0
Step 3/3 : EXPOSE 8080/tcp
 ---> Running in 5d60579ef1fc
Removing intermediate container 5d60579ef1fc
 ---> 50126aaa54f2
Successfully built 50126aaa54f2
Successfully tagged test:latest
```
</details>

Example: expose before
-------------------------------

Observe:
- Expose layer now uses cache (` ---> Using cache`)
- Expose layer gets same ID on each invocation

Same layer ID means that Docker image transfers get done faster when redundant layers don't have to be transferred again every time a new version of the image gets published. Just the actually different layers need to be transferred.

<details>
<summary>Show</summary>

```
λ cat Dockerfile
FROM scratch

EXPOSE 8080/tcp

ADD date /date


/tmp/dockertest
λ echo $(date) > date && docker build -t test .
Sending build context to Docker daemon  3.072kB
Step 1/3 : FROM scratch
 --->
Step 2/3 : EXPOSE 8080/tcp
 ---> Running in af46eb388183
Removing intermediate container af46eb388183
 ---> f9b5c04ca0fd
Step 3/3 : ADD date /date
 ---> f2cd69027a97
Successfully built f2cd69027a97
Successfully tagged test:latest

/tmp/dockertest
λ echo $(date) > date && docker build -t test .
Sending build context to Docker daemon  3.072kB
Step 1/3 : FROM scratch
 --->
Step 2/3 : EXPOSE 8080/tcp
 ---> Using cache
 ---> f9b5c04ca0fd
Step 3/3 : ADD date /date
 ---> f637537a22d5
Successfully built f637537a22d5
Successfully tagged test:latest
```
</details>

**Review by joonas-fi — APPROVED** [review]
LGTM (though please see review comments for future tips)

File: docker_image_build.sh (hunk 1 above)
Comment (jnippula) [reply docker_image_build.sh:1]: Right, those scripts were for my own testing only, they are accidentally taken in. Definitely not needed to be added here

**Review by joonas-fi — APPROVED** [review]

**Comment by jnippula** [comment]
I renamed Dockerfile as Dockerfile.containerize to separate from debian package building.

File: .github/workflows/tii-agent-protocol-splitter.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+name: tii-agent-protocol-splitter
+
+on:
+  push:
+    branches: master
+  pull_request:
+    branches: [ master ]
+    tags:
```
Comment (kulmesa) [inline .github/workflows/tii-agent-protocol-splitter.yaml:8]: tags is in wrong place. It should be under push

**Review by kulmesa — CHANGES_REQUESTED** [review]

File: .github/workflows/tii-agent-protocol-splitter.yaml (hunk 3 above)
Comment (jnippula) [reply .github/workflows/tii-agent-protocol-splitter.yaml:6]: Done

**Pull request merged**